src/Features/Core/Portable/Ambiguity/AbstractAmbiguousTypeCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+            var diagnosticNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            if (!syntaxFacts.IsIdentifierName(diagnosticNode))
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbolInfo = semanticModel.GetSymbolInfo(diagnosticNode, cancellationToken);
+            if (SymbolCandidatesContainsSupportedSymbols(symbolInfo))
+            {
+                var addImportService = document.GetLanguageService<IAddImportsService>();
+                var diagnostic = context.Diagnostics.First();
+                var compilation = semanticModel.Compilation;
+                var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var placeSystemNamespaceFirst = optionSet.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, document.Project.Language);
+                var typeName = GetAliasFromDiagnosticNode(syntaxFacts, diagnosticNode);
+                foreach (var symbol in symbolInfo.CandidateSymbols)
+                {
+                    var aliasDirective = GetAliasDirective(typeName, symbol);
+                    var codeActionPreviewText = await GetTextPreviewOfChangeAsync(aliasDirective,
+                                                                                  document.Project.Solution.Workspace,
+                                                                                  optionSet,
+                                                                                  cancellationToken).ConfigureAwait(false);
+                    var newRoot = addImportService.AddImport(compilation, root, diagnosticNode, aliasDirective, placeSystemNamespaceFirst);
+                    var codeAction = new MyCodeAction(codeActionPreviewText, c => Task.FromResult(document.WithSyntaxRoot(newRoot)));
+                    context.RegisterCodeFix(codeAction, context.Diagnostics.First());
+                }
+            }
+        }
+
+        private static async Task<string> GetTextPreviewOfChangeAsync(SyntaxNode newNode, Workspace workspace, OptionSet optionSet, CancellationToken cancellationToken)
+        {
+            var formattedNode = await Formatter.FormatAsync(newNode, workspace, optionSet, cancellationToken).ConfigureAwait(false);
+            var formattedText = formattedNode.ToFullString();
+            return string.Format(FeaturesResources.Alias_ambiguous_type_0, formattedText);
+        }
+
+        private static string GetAliasFromDiagnosticNode(ISyntaxFactsService syntaxFacts, SyntaxNode diagnosticNode)
+        {
+            // The content of the node is a good candidate for the alias
+            // For attributes VB requires that the alias ends with 'Attribute' while C# is fine with or without the suffix.
```

Comment:
if only VB needs this, then why is this code running for both C# and VB?

Reply:
I changed the logic to rely on symbol.Name instead of the text of the IdentifierNameSyntaxNode. This way it is easier and also covers the attribute requirements of VB.